src/main/kotlin/org/rust/ide/refactoring/generate/BaseGenerateHandler.kt

```diff
@@ -86,7 +89,7 @@ abstract class BaseGenerateHandler : LanguageCodeInsightActionHandler {
             allowEmptySelection
         ) ?: return
         runWriteAction {
-            performRefactoring(context.struct, context.implBlock, chosenFields, context.substitution, editor)
+            performRefactoring(context.struct, context.implBlocks.firstOrNull(), chosenFields, context.substitution, editor)
```

Comment:
Is it better to place elements close to existing implementations when there are multiple implementation blocks? But this requires more work.

> Is it better to place elements close to existing implementations when there are multiple implementation blocks? 

This requires counting the number of related implementations that already exist in different blocks, which means that sorting is also required.

Maybe not a good idea.

Reply:
This currently causes the test to fail, as the order cannot be determined so the refactoring blocks are random.

Reply:
ping @Undin